File: carrito_v4/src-gen/windows/Common/ApiInterface.cs

//Start of user code imports
using Kulman.WPA81.BaseRestService.Services.Abstract;
using CarritoDeCompras.Model;
using System;
using System.Threading.Tasks;
using System.Collections.Generic;
//End of user code

namespace CarritoDeCompras.Common
{
    class ApiInterface : BaseRestService
    {
        protected override string GetBaseUrl()
        {
            return "http://www.api2.cart.com.py";
        }

		public Task<Boolean> login( User user)  
        {
			return Post<Boolean>("login" , user );
		}

		public Task<Boolean> logout()  
        {
			return Post<Boolean>("logout" , null);
		}

		public Task<List<Product>> getAllProducts()  
        {
			return Get<List<Product>>("products");
		}

		public Task<Product> getProduct()  
        {
			return Get<Product>("products");
		}

		public Task<List<Provider>> getAllProviders()  
        {
			return Get<List<Provider>>("provider");
		}

		public Task<Provider> getProvider()  
        {
			return Get<Provider>("provider");
		}

		public Task<Boolean> checkOut( Product products)  
        {
			return Post<Boolean>("cart/confirm" , products );
		}

		public Task<List<ShoppingCart>> getUserCart()  
        {
			return Get<List<ShoppingCart>>("cart");
		}
    }
}

File: carrito_v4/src-gen/android/app/src/main/java/com/example/carritodecompras/conn/ApiInterface.java

package com.example.carritodecompras.conn;

//Start of user code imports
import java.util.List;

import retrofit2.Call;
import retrofit2.http.DELETE;
import retrofit2.http.GET;
import retrofit2.http.PATCH;
import retrofit2.http.POST;
import retrofit2.http.PUT;
//End of user code

public interface ApiInterface {
	@POST("login")
	Call<Boolean> login( User user );

	@POST("logout")
	Call<Boolean> logout();

	@GET("products")
	Call<List<Product>> getAllProducts();

	@GET("products")
	Call<Product> getProduct();

	@GET("provider")
	Call<List<Provider>> getAllProviders();

	@GET("provider")
	Call<Provider> getProvider();

	@POST("cart/confirm")
	Call<Boolean> checkOut( Product products );

	@GET("cart")
	Call<List<ShoppingCart>> getUserCart();
}
